From patchwork Wed Sep 26 02:42:39 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10615103
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 18F56112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Sep 2018 02:42:51 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id E7BD52AE64
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Sep 2018 02:42:50 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id DA3CB2AE66; Wed, 26 Sep 2018 02:42:50 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 765BD2AE64
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Sep 2018 02:42:50 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726171AbeIZIxZ (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Sep 2018 04:53:25 -0400
Received: from mail-pf1-f193.google.com ([209.85.210.193]:38081 "EHLO
        mail-pf1-f193.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725901AbeIZIxZ (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Sep 2018 04:53:25 -0400
Received: by mail-pf1-f193.google.com with SMTP id x17-v6so12551565pfh.5
        for <kvm@vger.kernel.org>; Tue, 25 Sep 2018 19:42:48 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id;
        bh=Kk801CB7HECsIfI55iachf+GI5wWO6wdg4izsrGRZiA=;
        b=qkEeYPW48FIFacv9Hm3PhUfd2zSS9CKymcAfv1BJIP7ikmawbacE8kVCOX0Wl6pX4E
         WaV07Jnv6hXex+uHR9NZfSyNiIFu3iyEv7SqZvXbXe6D6IlwfrNGSSg7LRmrDvWD8zp6
         D5HJBkzWsJu8/vTX7xU2TPl/1mxoHUWiWJa+QpHEffEl7Q/M4DXA7Xh/wFZBbujvtLE7
         RiD8hUDZJl13tL1Jcog9vBaUHQYAPSnmkjGOB30yPw+y5zlyZVNshbhYgUJ87XMgxE25
         r5Suhs2+5oCC/iplm1/N7+LN2Vq9PgiMqkEkWvUmO0hQjVyRFJNmlQa4fFl5pF8NAcA8
         doMw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=Kk801CB7HECsIfI55iachf+GI5wWO6wdg4izsrGRZiA=;
        b=jT6PVyeM2UtbQq9RDzj/f1X0He2w3gOVHFAmrlrce7Vw9HukSSzY1/8S/O6R8mUri4
         wdOmFmSSI7Gaggb4l2xYSfVUwmDYYKG2C9eJuCFGuIW9AyvKXEcUwXXpZNQ7yyQ0vLCw
         4Yz7P6n6KGoputk0urvKXMT+V3pVgUY+4AvxTLhvyNDK+6bk3wuLNEwK53tWZ7+abqv4
         4BmnTEKYs1JplKMWFKM1+zDH9KM3D521/lt8/K0Rfyn6AtHmjPINYttCP8c9oPtHvYJF
         ZtOhKI4hin8swJ3kBVpFoHnjUno6myfcdAMc1TaW7KWBZclKCQ110jDtsHwcx+LAb5Br
         QzRQ==
X-Gm-Message-State: ABuFfojsQXGLxn1iGoqAkHeJ5pDGyWnLYYNPdzyG+l55U9xoC4ee7Zu3
        wC9oSU17YzwGvXrNgvBKZ6s=
X-Google-Smtp-Source: 
 ACcGV63j/djXeuqoU/UGpsj+rXABNGTURsoZpXkf0Zdt1R0xUJlR18uY9rdunGGfdhu/wufBdqum1Q==
X-Received: by 2002:a62:cac5:: with SMTP id
 y66-v6mr3883054pfk.187.1537929767763;
        Tue, 25 Sep 2018 19:42:47 -0700 (PDT)
Received: from localhost ([185.92.221.13])
        by smtp.gmail.com with ESMTPSA id
 n26-v6sm4348037pgv.78.2018.09.25.19.42.46
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Tue, 25 Sep 2018 19:42:47 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: x86@kernel.org, kvm@vger.kernel.org,
        Wei Yang <richard.weiyang@gmail.com>
Subject: [PATCH] KVM: x86: remove check on nr_mmu_pages in
 kvm_arch_commit_memory_region()
Date: Wed, 26 Sep 2018 10:42:39 +0800
Message-Id: <20180926024239.26048-1-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

* nr_mmu_pages would be non-zero only if kvm->arch.n_requested_mmu_pages is
  non-zero.

* nr_mmu_pages is always non-zero, since kvm_mmu_calculate_mmu_pages()
  never return zero.

Based on these two reasons, we can merge the two *if* clause and use the
return value from kvm_mmu_calculate_mmu_pages() directly.

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
Reviewed-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 arch/x86/kvm/x86.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 8614199d733b..7efd0ec7abd3 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -9131,13 +9131,9 @@ void kvm_arch_commit_memory_region(struct kvm *kvm,
 				const struct kvm_memory_slot *new,
 				enum kvm_mr_change change)
 {
-	int nr_mmu_pages = 0;
-
 	if (!kvm->arch.n_requested_mmu_pages)
-		nr_mmu_pages = kvm_mmu_calculate_mmu_pages(kvm);
-
-	if (nr_mmu_pages)
-		kvm_mmu_change_mmu_pages(kvm, nr_mmu_pages);
+		kvm_mmu_change_mmu_pages(kvm,
+					 kvm_mmu_calculate_mmu_pages(kvm));
 
 	/*
 	 * Dirty logging tracks sptes in 4k granularity, meaning that large
